Training/pointers.cpp

```diff
@@ -21,7 +21,7 @@ int hackerrank::pointers()
     int a, b;
     int *pa = &a, *pb = &b;
     
-    scanf("%d %d", &a, &b);
+    scanf("%d %d", pa, pb);
```

Comment:
So-- one of the unfortunate realities of C++ is that `template` appropriate code has snuck into C++ mainstream style. `address-of &` and `dereference *` operators can be _overloaded_ to offer different behaviors. The idiom in C++ for `iterators` involves overloading `&, *, ++, --` in such a way that they may actually not be inexpensive operations. So as a habit, you should aim to minimize their use.

Even here, the operator use here is obnoxiously inexpensive, but not necessarily free. You've already got the reference, so I was just pointing out that you _could_ go ahead and use it.

Reply:
I see what you are saying about the non-zero cost of using the address-of operator but it wasn't done that way in this case because the `scan` and `print` statements in this function are trying to prove a point: that storing values as they were stored previously results in access that is the same through pointers. Using the pointer variables in the `scan` here kind of undoes that intent.

Again this is one of the situations where tutorial code is harder to review.